Services/MovieDataService.swift

```diff
@@ -13,61 +13,87 @@ protocol MovieDataServiceProtocol {
+    var isPlayingNowRequestCompleted = true
+    var isMostPopularRequestCompleted = true
+    
     func getPlayingNowMoviesList(completion: @escaping (Result<[Movie], MovieServiceError>)-> Void) {
-        guard let url = URL(string: "https://api.themoviedb.org/3/movie/popular?api_key=b2b14caf40262a9c19a366b15e4e3537&language=en-US&page=1") else { return }
-
+        guard isPlayingNowRequestCompleted else { return }
+        guard let url = URL(string: "https://api.themoviedb.org/3/movie/popular?api_key=b2b14caf40262a9c19a366b15e4e3537&language=en-US&page=\(playingNowPage)") else { return }
+        
+        isPlayingNowRequestCompleted = false
         let request = URLRequest(url: url)
         URLSession.shared.dataTask(with: request) {(data, response, error) in
             if let error = error {
                 print(error.localizedDescription)
-//                How can I use error.localizedDescription to display the error in the VC?
-                completion(.failure(.errorOccurred))
+                completion(.failure(.failedToGetResponse))
                 return
             }
-            
+                   
             guard let data = data else {
                 completion(.failure(.failedToGetData))
                 return
             }
+             
+            self.isPlayingNowRequestCompleted = true
             
             let jsonDecoder = JSONDecoder()
             guard let moviesData = try? jsonDecoder.decode(Movies.self, from: data) else {
-                completion(.failure(.failedToDecodeData))
+                completion(.failure(.failedToDecode))
                 return
-            }
+                }
+            
+            self.playingNowPage += 1
             completion(.success(moviesData.results))
         }.resume()
     }
     
     func getMostPopularMoviesList(completion: @escaping (Result<[Movie], MovieServiceError>)-> Void) {
-        guard let url = URL(string: "https://api.themoviedb.org/3/movie/now_playing?api_key=94806a6f0ae52fc236885e625fc54d47&language=en-US&page=1") else { return }
+        guard isMostPopularRequestCompleted else { return }
+        guard let url = URL(string: "https://api.themoviedb.org/3/movie/now_playing?api_key=94806a6f0ae52fc236885e625fc54d47&language=en-US&page=\(mostPopularPage)") else { return }
 
+        isMostPopularRequestCompleted = false
         let request = URLRequest(url: url)
         URLSession.shared.dataTask(with: request) {(data, response, error) in
             if let error = error {
                 print(error.localizedDescription)
-                completion(.failure(.errorOccurred))
+                completion(.failure(.failedToGetResponse))
                 return
             }
-            
+                    
             guard let data = data else {
                 completion(.failure(.failedToGetData))
                 return
             }
             
+            self.isMostPopularRequestCompleted = true
```

Comment:
Same as previously, we need to set `isMostPopularRequestCompleted` back to true as soon as the data task is done, regardless of success or not